### PR TITLE
Updates to Travis CI

### PR DESCRIPTION
**Pull request by pietro**
I tried to make the commits not too broad in scope. I'm sorry I didn't see you were also working on re-enabling kcov. I should have filled an issue when I started to look into this.

### CODE REVIEW

File: mk/travis-install-kcov.sh [hunk 1]
```diff
@@ -24,9 +24,9 @@ set -ex
 
 # kcov 26 or newer is needed when getting coverage information for Rust.
 # kcov 31 is needed so `kcov --version` doesn't exit with status 1.
-KCOV_VERSION=${KCOV_VERSION:-31}
+KCOV_VERSION=${KCOV_VERSION:-36}
```
Comment (briansmith) [inline mk/travis-install-kcov.sh]: The comments above this line explain why kcov 31 is needed. Please document why kcov 36 is needed.

File: mk/update-travis-yml.py [hunk 2]
```diff
@@ -117,15 +122,15 @@ def format_entry(os, target, compiler, rust, mode, features):
     vendor = target_words[1]
     sys = target_words[2]
 
-    # Currently kcov only runs on Linux.
+    # Currently kcov only runs on Linux and MacOS.
```
Comment (briansmith) [inline mk/update-travis-yml.py]: NIT: "macOS", not "MacOS".

File: mk/update-travis-yml.py [hunk 3]
```diff
@@ -145,16 +150,12 @@ def prefix_all(prefix, xs):
         packages = sorted(get_linux_packages_to_install(target, compiler, arch, kcov))
         sources_with_dups = sum([get_sources_for_package(p) for p in packages],[])
         sources = sorted(list(set(sources_with_dups)))
-
-    # TODO: Use trusty for everything?
-    if arch in ["aarch64", "arm", "armv7"]:
         template += """
-      dist: trusty
-      sudo: required"""
+      dist: trusty"""
```
Comment (briansmith) [inline mk/update-travis-yml.py]: I think this can just be prepended to `entry_linux_packages_template`

File: mk/update-travis-yml.py [hunk 4]
```diff
@@ -145,16 +150,12 @@ def prefix_all(prefix, xs):
...
 
     if sys == "linux":
```
Comment (briansmith) [inline mk/update-travis-yml.py]: This `if` can be combined with the previous `if`.

File: mk/travis-install-kcov.sh (hunk 1 above)
Comment (pietro) [reply mk/travis-install-kcov.sh]: That should have been in the macos commit. I'll move it and add the comment.

File: mk/travis.sh [hunk 5]
```diff
@@ -104,8 +116,15 @@ if [[ "$KCOV" == "1" ]]; then
   RUSTFLAGS="-C link-dead-code" \
     cargo test -vv --no-run -j2  ${mode-} ${FEATURES_X-} --target=$TARGET_X
   mk/travis-install-kcov.sh
+
+  # Add `/usr/bin` to the front of PATH so MacOS will use the system python.
```
[the remark this answers is not in the document]
Comment (pietro) [reply mk/travis.sh]: Kcov uses the XCode LLDB. I don't know why but LLDB loads some python scripts When I don't change the PATH it uses the python travis installs, via homebrew I think, and it spits out a lot of python import error messages and the build seem to hang.

**Comment by briansmith** [comment]
I merged the first two commits of this already:
59b550c07465e5312be2bbce5be5862cd930c356
0f246702f11866f8664b6af0d25794c41ad6b135

**Comment by pietro** [comment]
@briansmith should I keep looking into adding kcov to macos builds? The kcov macos build for this PR timeout but I haven't had this issue on my builds after the PATH hack.

**Comment by briansmith** [comment]
> @briansmith should I keep looking into adding kcov to macos builds? The kcov macos build for this PR timeout but I haven't had this issue on my builds after the PATH hack.

Yes, I think it would be great to get the code coverage on macOS too. I didn't marge the macOS changes in this PR yet only because I want to have the aforementioned nits addressed first.

File: mk/travis-install-kcov.sh (hunk 1 above)
Comment (pietro) [reply mk/travis-install-kcov.sh]: Comment added.

File: mk/travis.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (pietro) [reply mk/travis.sh]: I wrapped a `if [[ "$TARGET_X"  =~ "apple" ]]` around the PATH changes.

File: mk/update-travis-yml.py (hunk 2 above)
Comment (pietro) [reply mk/update-travis-yml.py]: Done

File: mk/update-travis-yml.py (hunk 3 above)
Comment (pietro) [reply mk/update-travis-yml.py]: I kept it separate to make it easier to be able to use a different distro.

File: mk/update-travis-yml.py (hunk 4 above)
Comment (pietro) [reply mk/update-travis-yml.py]: I merged it with the previous and the next if.

**Comment by pietro** [comment]
@briansmith I marked all the conversations as resolved. Let me know if there's something else you'd like me to change.

**Comment by briansmith** [comment]
Thanks again! Merged!

**Comment by briansmith** [comment]
Sorry, I had to revert this in b64f34742453dba41479195d0192e5d07c13f1c5 since every maOS debug build after it merged was broken.

**Comment by briansmith** [comment]
I suggest we try relanding this, but just don't run it on Travis CI. That way people can run the script locally on macOS.

**Comment by pietro** [comment]
>>> I suggest we try relanding this, but just don't run it on Travis CI. That way people can run the script locally on macOS.

I'll rip the part that add `KCOV=1` to macos builds but I'll leave the comments about it on.

**Pull request merged**